donut/modules/calendar/static/script.js

```diff
@@ -0,0 +1,388 @@
+    $("div").each( function (){
+        if ($(this).hasClass(tag))
+        {
+            $(this).toggle(visibility);
+        }
+    });
+}
+
+
+// Returns the day of week which month starts (eg 0 for Sunday, 1 for Monday, etc.)
+function getCalendarStart(dayOfWeek, currentDate) {
+  var date = currentDate - 1;
+  var startOffset = (date % 7) - dayOfWeek;
+  if (startOffset > 0) {
+    startOffset -= 7;
+  }
+  return Math.abs(startOffset);
+}
+
+function showEventInfo(event)
+{
+  data = event.data.data;
+  $('#eventName').text(data.summary);
+  $('#eventName_edit').val(data.summary);
+ 
+  var begin = data.start.dateTime || data.start.date;
+  var end = data.end.dateTime || data.end.date;
+  $('#eventStartTime').text(begin);
+  $('#eventEndTime').text(end);
+  
+  $('#eventDescription').text(data.description);
+  $('#eventDescription_edit').val(data.description);
+
+
+  $('#eventLocation').text(data.location);
+  $('#eventLocation_edit').val(data.location);
+
+  $('#tag').text(data.organizer.displayName);
+  $('#htag').val(data.organizer.displayName);
+  $('#calEventId').val(data.id);
+  // We force a start time on our site but google doesn't require it
+  if(begin.length > 10)
+  {
+    $('#start_hour').val(begin.substring(11, 13));
+    $('#end_hour').val(end.substring(11, 13));
+    $('#start_minute').val(end.substring(14, 16));
+    $('#end_minute').val(end.substring(14, 16));
+  }
+  else
+  {
+    $('#start_hour').val(0);
+    $('#end_hour').val(0);
+    $('#start_minute').val(0);
+    $('#end_minute').val(0); 
+  }
+  $('#start_date').val(begin.substring(0, 10));
+  $('#end_date').val(end.substring(0, 10));
+  $(".deleteEvent").off("click");
+  $(".deleteEvent").on('click',{eventID:data.id, tag:data.organizer.displayName}, deleteEvent);
+  
+
+
+  // Rendering fun....
+  var eleCenter = $('#extraInfo_triangle').width()/2;
+  if (event.clientX < $('#cal').width()/2)
+  {
+    $('#extraInfo_triangle').css("left", $('#cal').width()/2 - eleCenter);
+    $('#extraInfo').css("left", $('#cal').width()/2);  
+    $('#extraInfo_triangle').css({"box-shadow": "3px -3px 3px 0px rgba(0, 0, 0, 0.4)"});
+    $('#extraInfo_triangle').css({"border-color": "white white transparent transparent"});
+  }
+  else {
+    $('#extraInfo').css("left", "0px");
+    $('#extraInfo_triangle').css("left", $('#cal').width()/2 - eleCenter);
+    $('#extraInfo_triangle').css({"box-shadow": "-3px 3px 3px 0px rgba(0, 0, 0, 0.4)"});
+    $('#extraInfo_triangle').css({"border-color": "transparent transparent white white"});
+  }
+  $('#extraInfo_triangle').css("top", 
+        Math.min($('#extraInfo').height() -  $('#extraInfo_triangle').height(), event.clientY+ 5));
+  $('#extraInfo_triangle').fadeIn();
+  $('#extraInfo').fadeIn();
+}
+
+// Render the events for a month and year on our calendar. 
+function renderEvents(month, year)
+{
+  $('div').filter($('.day')).each(function(){$(this).empty();});
+  for(var i = 0; i < events[month].length; i++)
```

Comment:
If you aren't using the index `i`, could do `events[month].forEach(function(curEvent) {...})`